app/src/main/java/com/android/shelfLife/model/newhousehold/HouseHoldRepositoryFirestore.kt

```diff
@@ -111,37 +121,60 @@ class HouseholdRepositoryFirestore(
    * @param household The household with updated data.
    */
   override suspend fun updateHousehold(household: HouseHold) {
+    var originalItem: HouseHold? = null
```

Comment:
You add the originalItem to store the information of the household you are in (and are going to leave). We store this in case there is an issue with the creation of another / moving to another and we can rollback safely. 
"Very nice" ~Borat